### ISSUE TITLE
Tables, coloured cells and WCAG

### ISSUE THREAD

**Issue by juliemercier** [wet-boew/wet-boew#7590]
The [ coloured cells section of the tables page](http://wet-boew.github.io/wet-boew-styleguide/v4/design/tables-en.html#tables-contextual-classes) does not meet **WCAG 1.41. Use of color**:  _Color is not used as the only visual means of conveying information, indicating an action, prompting a response, or distinguishing a visual element. (Level A)_ .
Additional information needs to be added to coloured table cells to indicate the meaning of the colour to screen reader users . I want to use the `<td class="danger">` to represent protected data (in a protected environment) and need to add `<span class="wb-inv">Protected B data</span>` in the cell preceding the protected data value. Perhaps a reminder not to use the coloured cells without an indicator for screen readers could be included in the documentation? 
Would a visual element need to be added for colour-blind users? If  a colour-blind user cannot distinguish between the red background and a green background, then would there not need to be another visual cue or very specific text to differentiate one from the other?

**Comment by shawnthompson** [comment]
@juliemercier I agree with you, care to write something up or even make a PR for the change?

**Comment by juliemercier** [comment]
How about: In order to meet **WCAG 1.4.1 Use of color** when using coloured cells, remember to provide screen reader users and colour-blind users with additional cues as to the meaning of the colour.

**Comment by byshine** [comment]
@juliemercier I agree, this may also affect `<table class="table table-hover">`

**Comment by hsrudnicki** [comment]
Does this remain still an issue or otherwise? Please update the status of this again, whether this can be closed or otherwise, thx

**Comment by LaurentGoderre** [comment]
It doesn't faill 1.4.1 because the color is there to reinforce with the content of the cell (ex: Successful or positive action, Warning that might need attention, Dangerous or potentially negative action)

**Comment by cybersubtype** [comment]
I think @dfarough would agree that the color of the cells only works visually. In some cases the colors aren't conveyed to screen readers. Also see issue #7662.

**Comment by rlambert27** [comment]
I would agree that the example provided would fail 1.4.1. I think @juliemercier is right that a note should be added to tell authors to not only use color as a visual means of indicating something special within a cell or row.

However the table-hover does not fail 1.4.1 because the user is doing the highlighting, not the content author wanting to highlight something visually.

**Comment by LaurentGoderre** [comment]
It's a weird example because it depends on context but in this case it doesn't fail because of the text representation of what the color means, though it would be easy to implement it incorrectly

**Comment by rlambert27** [comment]
Well the example for the red color says "Dangerous or potentially negative action". The only way to know that something in this cell is "Dangerous or potentially negative action" is the red background color. I don't think this passes G14 (https://www.w3.org/TR/WCAG20-TECHS/G14.html) technique and no other technique seems to be implemented.

**Comment by masterbee** [comment]
@juliemercier this is really good discussion but I am not sure it belongs in this repo or the repo that you linked to as the example. This to me illustrates a good style guide issue and has healthy dialogue as well. 

Can you post over in the style-guide repo since this will allow for the maintainers of the style-guide to be aware of this thread.

**Comment by masterbee** [comment]
closed issue since asked creator to submit issue to to wet-boew-style-guide repo
